### PR TITLE
fix(gatsby-source-wordpress): bug patch for issue 35460

### LINKED ISSUE #35460: [gatsby-source-wordpress] Cannot destructure property 'typeInfo' of 'queryInfo' as it is undefined.
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

i'm getting the following error during development: 

```
TypeError: Cannot destructure property 'typeInfo' of 'queryInfo' as it is undefined.
    at fetchReferencedMediaItemsAndCreateNodes (.../node_modules/gatsby-source-wordpress/src/steps/source-nodes/fetch-nodes/fetch-referenced-media-items.js:528:11)
```

I am assuming it is because I removed the `RootQuery.mediaItems` type from WPGraphQL (due to not having a reliable way to tell `gatsby-source-wordpress` not to source it which is drastically slowing down my build times and I don't need the `mediaItems` on my gatsby site) via: 

```php
add_filter( 'graphql_RootQuery_fields', function( $fields ) {   	 	
  unset($fields['mediaItems']);
  return $fields;
}, 99, 1 );
```

Can this be changed to check for `undefined` first and/or create a plugin option to ignore certain nodes entirely on the `RootQuery`?

### Reproduction Link

https://github.com/tsdexter/gatsby-repro

### Steps to Reproduce

1. run `npm run develop` 
2. change wordpress data
3. develop process crashes

In order to run it with your own WP instance you will likely need to add to your functions.php file:

```
add_filter( 'graphql_NodeWithFeaturedImage_fields', function( $fields ) {
		unset($fields['featuredImage']);
    return $fields;
}, 99, 1 );

add_filter( 'graphql_User_fields', function( $fields ) {
		unset($fields['mediaItems']);
    return $fields;
}, 99, 1 );

add_filter( 'graphql_RootQuery_fields', function( $fields ) {   	 	
		unset($fields['mediaItems']);
    return $fields;
}, 99, 1 );
```

as it's likely the related code thats breaking it. 

### Expected Result

build process shouldn't crash

### Actual Result

build process crashes with error message

### Environment

```shell
System:
    OS: macOS 12.2
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 100.0.4896.127
    Firefox: 97.0.2
    Safari: 15.3
  npmPackages:
    gatsby: ^4.10.0-next.3 => 4.10.0-next.3 
    gatsby-plugin-fastify: ^0.7.4 => 0.7.4 
    gatsby-plugin-image: ^2.10.0-next.3 => 2.10.0-next.3 
    gatsby-plugin-sharp: ^4.10.0-next.3 => 4.10.0-next.3 
    gatsby-source-wordpress: ^6.9.1 => 6.9.1 
    gatsby-transformer-sharp: ^4.10.0-next.2 => 4.10.0-next.2 
  npmGlobalPackages:
    gatsby-cli: 4.12.1
```

### Config Flags

_No response_

### PR DESCRIPTION
**Pull request by tsdexter**
## Description

Check the `pluginOptions.type.MediaItem.exclude` before fetching referenced media items during page updates

## Related Issues

Fixes #35460

### CODE REVIEW

File: packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/wp-actions/update.js [hunk 1]
```diff
@@ -186,9 +186,11 @@ export const createSingleNode = async ({
     helpers,
   })
 
-  await fetchReferencedMediaItemsAndCreateNodes({
-    referencedMediaItemNodeIds: nodeMediaItemIdReferences,
-  })
+  if (!pluginOptions.type.MediaItem.exclude) {
```
Comment (TylerBarnes) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/wp-actions/update.js:189]:
```suggestion
  if (!pluginOptions.type?.MediaItem?.exclude) {
```

**Review by TylerBarnes — CHANGES_REQUESTED** [review]

**Review by TylerBarnes — APPROVED** [review]
Looks good! Thanks for this @tsdexter !

**Comment by tsdexter** [comment]
@TylerBarnes would it make sense to localize the check into the function? If so, I can do that and cleanup the other individual checks. 

Something like: 

```js
export default async function fetchReferencedMediaItemsAndCreateNodes({
  referencedMediaItemNodeIds,
  mediaItemUrls,
}) {
  const state = store.getState()
  const queryInfo = state.remoteSchema.nodeQueries.mediaItems
  const { helpers, pluginOptions } = state.gatsbyApi

  if (!pluginOptions.type.MediaItem.exclude) return []

  const { createContentDigest, actions } = helpers
  const { url } = pluginOptions
  const { typeInfo, settings, selectionSet, builtFragments } = queryInfo
...
```

**Comment by TylerBarnes** [comment]
Yeah, that seems reasonable 👍

**Pull request merged**